MAUI/DataForm/Working-with-DataForm.md

```diff
@@ -0,0 +1,781 @@
+
+Here, `Salary` data field is restricted from being edited in the data form.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "MiddleName")
+            {
+                e.DataFormItem.IsReadOnly = true;
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+#### Changing DataFormItem visibility
+
+You can change the `DataFormItem` visibility by using the `IsVisible` property in the `DataFormItem`.
+
+Here, `LastName` data field will be hidden.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "LastName")
+            {
+                e.DataFormItem.IsVisible = false;
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Setting watermark
+
+You can display the watermark in the editor by defining the display attribute or using the`GenerateDataFormItem` event.
+
+#### Using attribute
+
+You can show the watermark in the editor by setting the `Prompt` in display attribute.
+
+{% tabs %}
+{% highlight C# %}
+
+    [Display(Prompt = "Enter middle name")]
+    public string MiddleName { get; set; }
+    
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+You can show the watermark in the editor by using the `PlaceholderText` property in `DataFormItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "Description")
+            {
+                e.DataFormItem.PlaceholderText = "Enter description";
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+## Setting watermark color
+
+You can display the color for the watermark in the editor by using `GenerateDataFormItem` event.
+
+You can change the color for the watermark in the editor by using the `PlaceholderColor` property in `DataFormItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "Description")
+            {
+                e.DataFormItem.PlaceholderText = "Enter description";
+                e.DataFormItem.PlaceholderColor = Colors.MistyRose;
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+## Setting Padding
```

Comment:
## Change padding for auto generated editor

You can create a space around a label and editor by using `Padding` property in the `DataFormItem` by using GenerateDataFormItem event when Autogerate true

Reply:
title and content updated